### ISSUE TITLE
Batch requests yield 400/404 errors with reason "Ok"

### ISSUE THREAD

**Issue by GoogleCodeExporter** [Dragon-Born/google-api-python-client#165]
```
My callback function for handling batch request errors is receiving HttpErrors 
with status 40X and reason "Ok" for requests to invalid or missing resources.

I've included a simple script that reproduces the issue. Simply run it in a 
directory with a credential storage file named 'credentials.json'.

This is not a server-side issue since printing headers and content in this for 
loop
  http://code.google.com/p/google-api-python-client/source/browse/apiclient/http.py#1111
reveals an appropriate server response.

What version of the product are you using? On what operating system?
version 1.0c2 on Ubuntu
```

Original issue reported on code.google.com by `yova...@google.com` on 11 Jul 2012 at 3:29

Attachments:
- [batch404ok.py](https://storage.googleapis.com/google-code-attachments/google-api-python-client/issue-165/comment-0/batch404ok.py)

**Comment by GoogleCodeExporter** [comment]
```
CL out for review: http://codereview.appspot.com/6355092/
```

Original comment by `jcgregorio@google.com` on 11 Jul 2012 at 4:10
- Changed state: **Accepted**

**Comment by GoogleCodeExporter** [comment]
```
This issue was closed by revision 25632fda0fef.
```

Original comment by `jcgregorio@google.com` on 11 Jul 2012 at 6:44
- Changed state: **Fixed**
